caproto/threading/client.py

```diff
@@ -594,12 +602,18 @@ def _retry_unanswered_searches(self):
                 continue
 
             t = time.monotonic()
-            # Listify just so we can count the number of items and print a more
-            # informative debug message.
+
+            # filter to just things that need to go out
+            def _construct_search_requests(items):
+                for search_id, it in items:
+                    yield ca.SearchRequest(it[0], search_id, 13)
+                    it[-1] = t
+
             with self._search_lock:
-                items = list(self.unanswered_searches.items())
-            requests = (ca.SearchRequest(name, search_id, 13)
-                        for search_id, (name, _) in items)
+                items = list((search_id, it)
+                             for search_id, it in self.unanswered_searches.items()
+                             if (t - it[-1]) > RETRY_SEARCHES_PERIOD / 2)
```

Comment:
Just curious -- what's the `/ 2` for? Is this is an expedient way to try to make sure that we retry _at least_ that often, even accounting for the time it takes to go through this loop?

Reply:
Yes.